stdlib/public/Concurrency/TaskSleepDuration.swift

```diff
@@ -142,4 +142,20 @@ extension Task where Success == Never, Failure == Never {
   ) async throws {
     try await clock.sleep(until: deadline, tolerance: tolerance)
   }
+  
+  /// Suspends the current task for the given duration.
+  ///
+  /// If the task is canceled before the time ends, this function throws 
+  /// `CancellationError`.
+  ///
+  /// This function doesn't block the underlying thread.
+  ///
+  ///       try await Task.sleep(for: .seconds(3))
+  ///
+  @available(SwiftStdlib 5.7, *)
+  public static func sleep(
+    for duration: Duration
+  ) async throws {
+    try await sleep(until: .now + duration, clock: .continuous)
```

Comment:
It's not clear to me from the docs when one would reach for a suspending clock, so I went with `.continuous`, but I may just not understand their uses well enough.

Reply:
so suspending will mean that if the machine is asleep then the clock makes no forward progress so for example

sleep for 5 seconds
3 seconds later the machine is suspended
10 seconds later the machine is resumed
sleep continues for 2 seconds

for the continuous clock it would work as such:
sleep for 5 seconds
3 seconds later the machine is suspended
10 seconds later the machine is resumed
sleep resumes because 5 seconds has elapsed

Reply:
my guess is that more often than not, this particular variant would be the latter (continuous as you had a intuition for). If developers want the alternate version then they can use the slightly more verbose one.

Reply:
> so suspending will mean that if the machine is asleep then the clock makes no forward progress so for example

That was my understanding, too, I just can't think through when a suspending clock would be preferred :)